src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -13,60 +14,58 @@ public void addProduct(Product product) {
-            }
-        }
+        double sum = productBasket.values().stream().flatMap(Collection::stream)
+                .mapToDouble(x -> x.getPrice())
+                .sum();
         return sum;
-}
+    }
 
-public void printProductBasket() {
-    double sum = calculateSumOfBasket();
-    int specialCounter = 0;
-    for (Map.Entry<String, LinkedList<Product>> product : productBasket.entrySet()) {
-        System.out.println(product.toString());
-        for (Product p : product.getValue()) {
-            if (p.isSpecial()) {
-                specialCounter++;
-            }
+    public void printProductBasket() {
+        double sum = calculateSumOfBasket();
+        long specialCount = calculateSpecialProducts();
+        Stream<Product> basket = productBasket.values().stream().flatMap(Collection::stream);
+        if (sum == 0) {
+            System.out.println("в корзине пусто");
+        } else {
+            basket.forEach(System.out::println);
+            System.out.println("Итого: " + sum);
+            System.out.println("Специальных товаров: " + specialCount);
         }
     }
-    if (sum == 0) {
-        System.out.println("в корзине пусто");
-    } else {
-        System.out.println("Итого: " + sum);
-        System.out.println("Специальных товаров: " + specialCounter);
+
+    private long calculateSpecialProducts() {
+        long specialCounter = productBasket.values().stream().flatMap(Collection::stream)
+                .filter(x -> x.isSpecial())
+                .count();
+        return specialCounter;
     }
-}
 
-public boolean searchProduct(String name) {
-    for (Map.Entry<String, LinkedList<Product>> product : productBasket.entrySet()) {
-        for (Product p : product.getValue()) {
-            if (p.getName().equalsIgnoreCase(name)) {
-                return true;
+    public boolean searchProduct(String name) {
+        for (Map.Entry<String, LinkedList<Product>> product : productBasket.entrySet()) {
+            for (Product p : product.getValue()) {
+                if (p.getName().equalsIgnoreCase(name)) {
+                    return true;
+                }
             }
         }
+        return false;
     }
-    return false;
-}
-
-public void clear() {
-    productBasket.clear();
-}
 
-public LinkedList<Product> removeThisProduct(String name) {
-    LinkedList<Product> removedProducts = new LinkedList<>();
-    if (productBasket.containsKey(name)) {
-        removedProducts.addAll(productBasket.get(name));
-        productBasket.remove(name);
+    public void clear() {
+        productBasket.clear();
     }
-    if (removedProducts.isEmpty()) {
-        System.out.println("Список пуст");
+
+    public LinkedList<Product> removeThisProduct(String name) {
+        LinkedList<Product> removedProducts = new LinkedList<>();
+        if (productBasket.containsKey(name)) {
+            removedProducts.addAll(productBasket.get(name));
+            productBasket.remove(name);
```

Comment:
в 2 раза меньше кода
```suggestion
            removedProducts.addAll(productBasket.remove(name));
```